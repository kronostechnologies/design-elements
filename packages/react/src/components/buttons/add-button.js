"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
var plus_svg_1 = __importDefault(require("feather-icons/dist/icons/plus.svg"));
var button_1 = require("./button");
var PlusIcon = styled_components_1.default(plus_svg_1.default)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  height: 1rem;\n  margin-right: 0.5rem;\n  width: 1rem;\n"], ["\n  height: 1rem;\n  margin-right: 0.5rem;\n  width: 1rem;\n"])));
var AddButton = function (_a) {
    var children = _a.children, disabled = _a.disabled, onClick = _a.onClick, buttonType = _a.buttonType;
    var handleClick = function () { onClick && onClick(); };
    return (<button_1.Button disabled={disabled} onClick={handleClick} buttonType={buttonType}>
        <PlusIcon />
            {children}
        </button_1.Button>);
};
exports.AddButton = AddButton;
var templateObject_1;
