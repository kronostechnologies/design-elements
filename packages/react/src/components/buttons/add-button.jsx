import React from 'react';
import styled from 'styled-components';
import PlusSign from 'feather-icons/dist/icons/plus.svg';

import { SECONDARY } from '../../constants';

import primaryStyle from './styles/primary.js';
import secondaryStyle from './styles/secondary.js';

import AbstractButton from './abstract-button';


const Button = styled(AbstractButton)`
  ${props => (props.type === SECONDARY ? secondaryStyle : primaryStyle)}
`;

const PlusIcon = styled(PlusSign)`
  height: 1rem;
  margin-right: 0.5rem;
  width: 1rem;
`;

const AddButton = ({ children, disabled, onClick, type }) => (
    <Button disabled={disabled} onClick={onClick} type={type}>
        <PlusIcon />
        {children}
    </Button>
);

export default AddButton;
