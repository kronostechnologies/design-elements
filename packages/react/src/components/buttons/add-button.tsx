import React, { ButtonHTMLAttributes, DetailedHTMLProps, ReactElement } from 'react';
import styled from 'styled-components';
import { useDeviceContext } from '../device-context-provider/device-context-provider';
import { Icon } from '../icon/icon';
import { Button } from './button';

type ButtonType = 'primary' | 'secondary' | 'tertiary';

type Type = DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>['type'];

const PlusIcon = styled(Icon)`
    margin-right: var(--spacing-1x);
`;

interface ButtonProps {
    /**
     * Visual style
     * @default primary
     * */
    buttonType: ButtonType;
    /**
     * Sets button type
     * @default submit
     */
    type?: Type;
    label?: string;
    disabled?: boolean;

    onClick?(): void;
}

export function AddButton({
    type = 'submit',
    buttonType,
    disabled,
    label,
    onClick,
}: ButtonProps): ReactElement {
    const { isMobile } = useDeviceContext();

    return (
        <Button
            type={type}
            buttonType={buttonType}
            onClick={onClick}
            disabled={disabled}
            label={label}
        >
            <PlusIcon name="plusSign" size={isMobile ? '24' : '16'} />
        </Button>
    );
}
