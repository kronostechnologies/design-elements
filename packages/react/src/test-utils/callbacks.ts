export function doNothing(): void {
    // do nothing
}
