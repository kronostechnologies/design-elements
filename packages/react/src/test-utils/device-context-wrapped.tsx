import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
