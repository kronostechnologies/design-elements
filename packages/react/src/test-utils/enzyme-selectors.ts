import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
