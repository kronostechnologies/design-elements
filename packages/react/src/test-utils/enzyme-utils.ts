import { ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';

export async function waitForComponentToPaint<P = {}>(
    wrapper: ReactWrapper<P>,
    amount = 0,
): Promise<void> {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, amount));
        wrapper.update();
    });
}

export function expectFocusToBeOn(element: ReactWrapper): void {
    expect(document.activeElement).toBe(element.getDOMNode());
}
