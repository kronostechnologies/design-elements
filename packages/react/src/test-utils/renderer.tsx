import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
