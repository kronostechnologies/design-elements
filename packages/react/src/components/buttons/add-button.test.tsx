import { mountWithProviders, renderWithProviders } from '@design-elements/test-utils/renderer';
import React from 'react';
import { AddButton } from './add-button';

describe('Add Button', () => {
    test('onClick callback is called when clicked', () => {
        const callback = jest.fn();
        const wrapper = mountWithProviders(
            <AddButton onClick={callback} buttonType="primary" label="Primary Button" />,
        );

        wrapper.find(AddButton).simulate('click');

        expect(callback).toHaveBeenCalledTimes(1);
    });

    test('onClick callback cannot be called when disabled', () => {
        const callback = jest.fn();
        const wrapper = mountWithProviders(
            <AddButton onClick={callback} buttonType="primary" disabled label="Disabled" />,
        );

        wrapper.find(AddButton).simulate('click');

        expect(callback).toHaveBeenCalledTimes(0);
    });

    test('Button is disabled (desktop)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="primary" onClick={jest.fn()} disabled label="Primary Button" />,
            'desktop',
        );

        expect(tree).toMatchSnapshot();
    });

    test('Button is disabled (mobile)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="primary" onClick={jest.fn()} disabled label="Primary Button" />,
            'mobile',
        );

        expect(tree).toMatchSnapshot();
    });

    test('Button has primary styles (desktop)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="primary" onClick={jest.fn()} label="Primary Button" />,
            'desktop',
        );

        expect(tree).toMatchSnapshot();
    });

    test('Button has primary styles (mobile)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="primary" onClick={jest.fn()} label="Primary Button" />,
            'mobile',
        );

        expect(tree).toMatchSnapshot();
    });

    test('Button has secondary styles (desktop)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="secondary" onClick={jest.fn()} label="Secondary Button" />,
            'desktop',
        );

        expect(tree).toMatchSnapshot();
    });

    test('Button has secondary styles (mobile)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="secondary" onClick={jest.fn()} label="Secondary Button" />,
            'mobile',
        );

        expect(tree).toMatchSnapshot();
    });

    test('Button has tertiary styles (desktop)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="tertiary" onClick={jest.fn()} label="Tertiary Button" />,
            'desktop',
        );

        expect(tree).toMatchSnapshot();
    });

    test('Button has tertiary styles (mobile)', () => {
        const tree = renderWithProviders(
            <AddButton buttonType="tertiary" onClick={jest.fn()} label="Tertiary Button" />,
            'mobile',
        );

        expect(tree).toMatchSnapshot();
    });
});
