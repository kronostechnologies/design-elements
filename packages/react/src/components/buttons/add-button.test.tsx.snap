// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Button has disabled styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  line-height: 1rem;
  min-height: 32px;
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0 > svg {
  color: inherit;
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
}

.c1:focus {
  outline: none;
}

.c1:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c1:hover {
  background-color: #003A5A;
  border-color: #003A5A;
}

.c1:disabled {
  background-color: #84C6EA;
  border-color: #84C6EA;
}

.c2 {
  margin-right: var(--spacing-1x);
}

<button
  class="c0 c1"
  disabled=""
  type="submit"
>
  <svg
    class="c2"
    color="currentColor"
    focusable="false"
    height="16"
    width="16"
  />
  Primary Button
</button>
`;

exports[`Add Button has mobile styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.53px;
  -moz-letter-spacing: 0.53px;
  -ms-letter-spacing: 0.53px;
  letter-spacing: 0.53px;
  line-height: 1.5rem;
  min-height: 48px;
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-3x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0 > svg {
  color: inherit;
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
}

.c1:focus {
  outline: none;
}

.c1:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c1:hover {
  background-color: #003A5A;
  border-color: #003A5A;
}

.c1:disabled {
  background-color: #84C6EA;
  border-color: #84C6EA;
}

.c2 {
  margin-right: var(--spacing-1x);
}

<button
  class="c0 c1"
  type="submit"
>
  <svg
    class="c2"
    color="currentColor"
    focusable="false"
    height="24"
    width="24"
  />
  Primary Button
</button>
`;

exports[`Add Button has primary styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  line-height: 1rem;
  min-height: 32px;
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0 > svg {
  color: inherit;
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
}

.c1:focus {
  outline: none;
}

.c1:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c1:hover {
  background-color: #003A5A;
  border-color: #003A5A;
}

.c1:disabled {
  background-color: #84C6EA;
  border-color: #84C6EA;
}

.c2 {
  margin-right: var(--spacing-1x);
}

<button
  class="c0 c1"
  type="submit"
>
  <svg
    class="c2"
    color="currentColor"
    focusable="false"
    height="16"
    width="16"
  />
  Primary Button
</button>
`;

exports[`Add Button has secondary styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  line-height: 1rem;
  min-height: 32px;
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0 > svg {
  color: inherit;
}

.c1 {
  background-color: #FFFFFF;
  border-color: #006296;
  color: #006296;
}

.c1:focus {
  outline: none;
}

.c1:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c1:hover {
  border-color: #003A5A;
  color: #003A5A;
}

.c1:disabled {
  border-color: #84C6EA;
  color: #84C6EA;
}

.c2 {
  margin-right: var(--spacing-1x);
}

<button
  class="c0 c1"
  type="submit"
>
  <svg
    class="c2"
    color="currentColor"
    focusable="false"
    height="16"
    width="16"
  />
  Secondary Button
</button>
`;

exports[`Add Button has tertiary styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  line-height: 1rem;
  min-height: 32px;
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0 > svg {
  color: inherit;
}

.c1 {
  background-color: transparent;
  border-color: transparent;
  color: #60666E;
}

.c1:focus {
  outline: none;
}

.c1:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #00629666;
}

.c1:hover {
  background-color: #DBDEE1;
  color: #000000;
}

.c1:disabled {
  background-color: transparent;
  color: #B7BBC2;
}

.c2 {
  margin-right: var(--spacing-1x);
}

<button
  class="c0 c1"
  type="submit"
>
  <svg
    class="c2"
    color="currentColor"
    focusable="false"
    height="16"
    width="16"
  />
  Tertiary Button
</button>
`;
